src/engine/QueryPlanner.cpp

```diff
@@ -2405,36 +2405,40 @@ auto QueryPlanner::applyJoinDistributivelyToUnion(const SubtreePlan& a,
 std::optional<std::tuple<size_t, size_t>>
 QueryPlanner::getJoinColumnsForTransitivePath(const JoinColumns& jcs,
                                               bool leftSideTransitivePath) {
+  // If there are more than two pairs of join columns, we have a graph
+  // variable. In that case, we compute the full transitive hull.
   if (jcs.size() > 2) {
     return std::nullopt;
   }
+
+  // The index in `jcs` of the transitive path side and the other side.
   auto transitivePathIndex = static_cast<size_t>(!leftSideTransitivePath);
   auto otherIndex = static_cast<size_t>(leftSideTransitivePath);
-  // Since this function is only called for unbound transitive paths, we know
-  // that any extra variables have to be the graph var.
+
+  // If there is one pair of join columns, then either exactly one side of the
+  // transitive path can be bound (and we return that pair), or the graph
+  // variable is bound (in which case we return `std::nullopt`).
   auto graphColIndex = TransitivePathBase::firstGraphOrPayloadColumnIndex();
   if (jcs.size() == 1) {
     size_t transitiveCol = jcs[0][transitivePathIndex];
     size_t otherCol = jcs[0][otherIndex];
-    // We can't bind if the only matching column is the graph col. The
-    // `TransitivePathImpl` doesn't support this.
     if (transitiveCol >= graphColIndex) {
       return std::nullopt;
     }
-    // Bind on non-graph column.
     return std::tuple{transitiveCol, otherCol};
   }
-  // Bind on two columns, one graph column and one other column.
+
+  // Otherwise, we have two pairs of join columns. Then one pertains to the
+  // graph variable and the other to one side of the transitive path. Return
+  // the pair that pertains to the transitive path.
```

Comment:
Both pairs pertain to the transitive path, we return the one that's not the graph variable.